### PR TITLE
Feat/resolve on connect

### PR DESCRIPTION
**Pull request by milkpirate**
Add option to resolve target server address on each incomming connection.

### CODE REVIEW

File: common.c [hunk 1]
```diff
@@ -319,6 +319,11 @@ int connect_addr(struct connection *cnx, int fd_from, connect_blocking blocking)
     res = getpeername(fd_from, from.ai_addr, &from.ai_addrlen);
     CHECK_RES_RETURN(res, "getpeername", res);
 
+    if (cfg.protocols.resolve_on_forward) {
```
Comment (yrutschle) [inline common.c:322]: I would expect this to be `if (cnx->proto->resolve_on_forward) { ` instead (I'm not sure what this does: fetch the config item of the first protocol? cfg.protocols is an array...)

File: sslh-main.c [hunk 2]
```diff
@@ -124,8 +124,13 @@ static void config_protocols()
     int i;
     for (i = 0; i < cfg.protocols_len; i++) {
         struct sslhcfg_protocols_item* p = &(cfg.protocols[i]);
-        if (resolve_split_name(&(p->saddr), p->host, p->port)) {
-            print_message(msg_config_error, "cannot resolve %s:%s\n", p->host, p->port);
+
+        if (
+            !cfg.protocols.resolve_on_forward &&
```
Comment (yrutschle) [inline sslh-main.c:129]: same here: `if (!p->resolve_on_forward)`

**Review by yrutschle — CHANGES_REQUESTED** [review]
good starting point :-)

File: common.c (hunk 1 above)
Comment (milkpirate) [reply common.c:322]: You're totally rigth... 5 min hacked together :/

**Review by milkpirate — COMMENTED** [review]
Changed as desired

**Comment by yrutschle** [comment]
thanks for the contribution

**Pull request merged**